src/basic/namespace-util.c

```diff
@@ -231,3 +231,25 @@ int userns_acquire(const char *uid_map, const char *gid_map) {
         return TAKE_FD(userns_fd);
 
 }
+
+bool in_same_namespace(pid_t pid1, pid_t pid2, namespace_type ns)
+{
+        const struct namespace_info *nsinfo = &namespace_info[ns];
+        char ns_path[STRLEN("proc//") + DECIMAL_STR_MAX(pid_t) + strlen(nsinfo->proc_path)];
+        struct stat ns_st1, ns_st2;
+
```

Comment:
for most our helpers functions that deal with pids and look stuff up in /proc/, we add special support for map pid 0 to our own PID, inspired by `kill()`...

i.e. add something like:

```c
if (pid1 == 0)
        pid1 = getpid_cached();
if (pid2 == 0)
        pid2 = getpid_cached();
```

Reply:
(would then allow you to simply the caller in the later patch to just pass 0)